docs/source/release_roadmap.org

```diff
@@ -0,0 +1,363 @@
+virtual machine is statically linked. Adding an extra layer of dynamic
+linkage will result in a further overall reduction in efficiency given
+that permission validation (and the Executor RFC), are to be
+implemented and thus all ISI execution will go through the WASM
+sub-process.
+
+Furthermore, any change in the API will be accompanied by an extra
+copy of the =iroha_data_model= crate embedded into the Iroha runtime
+in addition to the copies that will occupy space in the Executor. 
+
+As such a monthly release cadence is the preferred solution as it
+allowed for a stable communication channel with the SDKs and the
+community, further boosting the profile of the project.
+
+* Scenario 2: ISA refactor only
+
+Scenario 2 is the situation in which the Instruction set architecture
+is to be modified, but the smartcontract language is kept as WASM.
+
+In this case, it is possible to mitigate the issue of the growing
+number of binary versions of =iroha_data_model= and freeze the API
+across different blockchains.
+
+In this scenario the Iroha Special Instructions become a frozen
+semantically versioned guarantee that a certain operation has a
+certain effect. As such, smartcontracts which link against the same
+_resident_ library dynamically (or statically linked at compile time)
+become portable across blockchains, provided they can be executed with
+the provided virtual machine. Further, the data model becomes an
+implementation detail for both the clients which were used at the
+submission time of the instructions and the Executor. The parts of the
+data model which affect the communication of the Executor to the host
+node remain frozen and thus don't need to be updated. Because
+instructions no longer enforce any of the blockchain invariants, this
+is not restrictive and a natural method of extension is provided by
+extending the Executor with custom expressions.
+
+No further benefits.
+
+* Scenario 3: ISA refactor and change of smartcontract language
+
+Depending on which smartcontract language is realised, the technical
+limitations incurred by usage of WASM can either be eliminated or
+substantially reduced. The precise character of the reductions shall
+be outlined in the RFC. But the short version follows: standards with
+guaranteed backwards compatibility, such as Java, native C-ABI
+libraries, and an embedded scripting language all eliminate the need
+for hosting multiple versions of multiple machines for validation.
+This greatly simplifies deployment of new nodes, attracting more users
+in the public blockchain space, as well as reducing the operational
+costs to private blockchains.
+
+Removal of WASM, will definitely eliminate the need to implement
+shared-memory dynamic linkage, and instead use the standard-native
+protocol. C-ABI dynamic linkage is implemented by the operating
+system, resulting in the best outcome. For Java some bridging code may
+be needed. For =lua= and Python, the dynamic linkage is an
```

Comment:
I wonder if I should point out the existence of [LuaJIT](https://luajit.org/), which is heavily optimized for `x86` and `x86-64` platforms? If we're talking about Lua, we must consider maximizing its performance.

Reply:
Remind me of that when I create the WASM-away RFC.